iib/web/models.py

```diff
@@ -1435,3 +1435,97 @@ def validate_registry_auths(registry_auths):
             raise ValidationError(err_msg)
         if not all(k == 'auth' and isinstance(v, str) for (k, v) in auth_dict.items()):
             raise ValidationError(err_msg)
+
+
+class RequestCreateEmptyIndex(Request, RequestIndexImageMixin):
+    """An "create-empty-index" image build request."""
+
+    __tablename__ = 'request_create_empty_index'
+
+    id = db.Column(db.Integer, db.ForeignKey('request.id'), autoincrement=False, primary_key=True)
+    from_index_id = db.Column(db.Integer, db.ForeignKey('image.id'), nullable=False)
+    from_index = db.relationship('Image', foreign_keys=[from_index_id], uselist=False)
+    binary_image_id = db.Column(db.Integer, db.ForeignKey('image.id'), nullable=True)
+    binary_image = db.relationship('Image', foreign_keys=[binary_image_id], uselist=False)
+    _labels = db.Column('labels', db.Text, nullable=True)
+
+    __mapper_args__ = {
+        'polymorphic_identity': RequestTypeMapping.__members__['create_empty_index'].value
+    }
+
+    @property
+    def labels(self):
+        """Return the Python representation of the JSON labels."""
+        return json.loads(self._labels) if self._labels else None
+
+    @labels.setter
+    def labels(self, labels):
+        """
+        Set the labels column to the input labels as a JSON string.
+
+        If ``None`` is provided, it will be simply set to ``None`` and not be converted to JSON.
+
+        :param dict labels: the dictionary of the labels or ``None``
+        """
+        self._labels = json.dumps(labels, sort_keys=True) if labels is not None else None
+
+    @classmethod
+    def from_json(cls, kwargs, batch=None):
+        """
+        Handle JSON requests for the create-empty-index API endpoint.
+
+        :param dict kwargs: the JSON payload of the request.
+        :param Batch batch: the batch to specify with the request.
+        """
+        request_kwargs = deepcopy(kwargs)
+
+        if not request_kwargs.get('from_index'):
+            raise ValidationError('"from_index" must be specified')
+
+        if request_kwargs.get('from_index') and not isinstance(
```

Comment:
`if request_kwargs.get('from_index') is not `None` and not ...` ? This way you can get rid of #1482

Reply:
On second thoughts, you don't really need this validation as the base class `from_json` will check for validity of `from_index`.

You can simply change line 1482 to check if `from_index` is not None to make sure your code doesn't falter on line 748.

Also, you might have to add a check here to make sure the user doesn't provide `add_arches`, `overwrite_from_index` and `overwrite_from_index_token`. Otherwise, IIB will process them as optional params and consider them valid because of the base class `from_json` function.

Reply:
done

Reply:
Should it be an error or I pop it?

Reply:
done